src/main/java/com/google/flourbot/entity/action/Action.java

```diff
@@ -1,7 +1,7 @@
 package com.google.flourbot.entity.action;
 
-public abstract class Action {
-  public abstract ActionType getActionType();
-  public abstract String getSheetId();
-  public abstract String getSheetUrl();
+public interface Action {
+  public ActionType getActionType();
+  public String getSheetId();
```

Comment:
no matter if this is an interface or abstract class, the sheet specific methods don't belong at the Action level. What if we add DocAction?

Reply:
renamed!